model.py

```diff
@@ -6069,14 +6069,12 @@ def get(cls, _db, url, do_get=None, extra_request_headers=None,
             media_type = None
 
         # At this point we can create/fetch a Representation object if
-        # we don't have one already, or if the URL or media type we
-        # actually got from the server differs from what we thought
-        # we had.
+        # we don't have one already, or if the URL we actually got from 
+        # the server differs from what we thought we had.
         if (not usable_representation 
-            or media_type != representation.media_type
             or url != representation.url):
             representation, is_new = get_one_or_create(
-                _db, Representation, url=url, media_type=media_type)
+                _db, Representation, url=url)
```

Comment:
For standard ebooks, we don't get a media type back at all when we fetch book files, but we can set it later based on the media type in the link. If we created a new representation instead of using the existing one, we might not be able to set the media type correctly later due to the unique constraint on url + media type. 

In general, if we fetch a resource and get a media type that's not what we thought, it seems like we should update the existing representation for that url instead of creating a new one, since the url may have changed. I can't think of a case where we'd want to have multiple media types for the same url. The unique constraint would make more sense on url only instead of url + media type, but I'm afraid to make that change in case we already have things that violate it.